kosat-api/src/commonMain/kotlin/Kosat.kt

```diff
@@ -0,0 +1,156 @@
+     * Resets the solver to its initial state, removing all variables and
+     * clauses.
+     */
+    fun reset() {
+        numberOfVariables = 0
+        numberOfClauses = 0
+        solver = CDCL()
+        invalidateResult()
+    }
+
+    /**
+     * Return the number of variables in the solver.
+     */
+    fun numberOfVariables(): Int {
+        return numberOfVariables
+    }
+
+    /**
+     * Return the number of clauses in the solver.
+     */
+    fun numberOfClauses(): Int {
+        return numberOfClauses
+    }
+
+    /**
+     * Allocate a new variable in the solver.
+     */
+    fun newVariable() {
+        invalidateResult()
+        numberOfVariables++
+        solver.newVariable()
+    }
+
+    /**
+     * Add a new clause to the solver. The literals must be in DIMACS format
+     * (positive for variables, negative for negated variables, 1-based
+     * indexing). All variables must be defined with [newVariable] before adding
+     * clauses.
+     */
+    fun newClause(clause: Iterable<Int>) {
+        checkLiterals(clause)
+        invalidateResult()
+        numberOfClauses++
+        solver.newClause(Clause.fromDimacs(clause))
+    }
+
+    /**
+     * Add a new clause to the solver. The literals must be in DIMACS format
+     * (positive for variables, negative for negated variables, 1-based
+     * indexing). All variables must be defined with [newVariable] before adding
+     * clauses.
+     */
+    fun newClause(vararg literals: Int) {
+        newClause(literals.asIterable())
+    }
+
+    /**
+     * Solve the SAT problem with the given assumptions, if any. Assumptions are
+     * literals in DIMACS format.
+     */
+    fun solve(vararg assumptions: Int): Boolean {
+        return solve(assumptions.asIterable())
+    }
+
+    /**
+     * Solve the SAT problem with the given assumptions
+     */
+    fun solve(assumptions: Iterable<Int>): Boolean {
+        checkLiterals(assumptions)
+        result = solver.solve(assumptions.map { Lit.fromDimacs(it) })
+        return result == SolveResult.SAT
+    }
+
+    /**
+     * If the problem is SAT, return the model as a list of booleans. [solve]
+     * must be called before calling this method. The model is cached after the
+     * first call and reset when clause or variable is incrementally is added.
+     */
+    fun getModel(): List<Boolean> {
+        if (result == null) {
+            throw IllegalStateException("Model is not available before solving")
+        } else if (result != SolveResult.SAT) {
+            throw IllegalStateException("Model is not available if result is not SAT")
+        }
+
+        return solver.getModel().also { model = it }
+    }
+
+    /**
+     * If the problem is SAT, return the value of the given literal in the
+     * model. [solve] must be called before calling this method.
+     */
+    fun value(literal: Int): Boolean {
+        if (literal == 0) {
+            throw IllegalArgumentException("Literal must not be 0")
```

Comment:
This can be written via `require` (it throws `IllegalArgumentException`, contrary to `check` which throws `IllegalStateException`).